src/api/api-utils.js

```diff
@@ -215,4 +222,38 @@ export default class ApiUtils {
     const mediaInfoData = await this.executeWithConcurrency(this.api.getBatchMediaInfo, this.infoSize, mediaKeyArray);
     return mediaInfoData;
   }
+
+  async copyOneDescriptionFromOther(mediaItems) {
+    try {
+      const item = mediaItems[0];
+      const itemInfoExt = await this.api.getItemInfoExt(item.mediaKey);
+      if (!itemInfoExt.descriptionFull && itemInfoExt.other) {
+        // The Google Photos API doesn't allow the description to be identical
+        // to the "Other" field.  Adding leading or trailing spaces doesn't
+        // work - if you try this using the web app, it simply deletes the
+        // description, and if you set it using the API directly then it
+        // ignores the description at display time.  However it *does* work to
+        // add a zero-width space (U+200B) since that character is not
+        // considered to be whitespace.
+        const description = itemInfoExt.other + "\u200B";
+        await this.api.setItemDescription(item.dedupKey, description);
+        return [true];
+      }
+      return [false];
```

Comment:
```suggestion
      if (!itemInfoExt.other && itemInfoExt.descriptionFull) {
        return
      }
    // The Google Photos API doesn't allow the description to be identical
    // to the "Other" field.  Adding leading or trailing spaces doesn't
    // work - if you try this using the web app, it simply deletes the
    // description, and if you set it using the API directly then it
    // ignores the description at display time.  However it *does* work to
    // add a zero-width space (U+200B) since that character is not
    // considered to be whitespace.
    const description = itemInfoExt.other + '\u200B';
    return await this.api.setItemDescription(item.dedupKey, description); // setItemDescription return empty array in case of success
```

executeWithConcurrency checks is result is an array. setItemDescription (and other methods) retuns array in case of success

Reply:
Sorry, this was confusing, I added some comments to explain better why I did things this way.  This method is intended to a boolean indicating whether the description was copied, so that at the end of the operation it can log how many descriptions were actually copied.  Descriptions are only copied if the 'Other' field is non-empty and the current Google Photos description *is* empty (i.e., it will never overwrite an existing description).  Returning [false] when no description was copied makes it easier to log progress as we go along; even with the increased parallelism, this operation is quite slow compared with batch operations (e.g., copying 400 descriptions on 500 photos takes about a minute on my tests).

Please let me know if there is a better way to do this; I don't have any previous experience with userscript and not much with Javascript or web development either.